### PR TITLE
[SELS-TASK] FE-Pagination-Bar

### PR DESCRIPTION
**Pull request by ragg09-sunasterisk**
### Asana Link
https://app.asana.com/0/1203341829928748/1203341829928697/f

### Commands
```
cd backend
php artisan serve
cd frontend
npm start
```
### Pre-conditions
none

### Expected Output
- [X] be able to paginate pages, and still functional actions

### Notes
- since I dont have a task specific for logout, I include some updates about that function in this pull request because soon, I will be moving to User Side.

### Screenshots
Pagination

### CODE REVIEW

File: fe/src/components/Pagination.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+const Pagination = (props) => {
+  const pageNumber = [];
+
+  for (let i = 1; i <= Math.ceil(props.totalItems / props.itemsPerPage); i++) {
+    pageNumber.push(i);
+  }
```
Comment (escbooster12-dev) [inline fe/src/components/Pagination.jsx]: - ahh you don't need a loop for this one. just get the `totalPageItems`
- and then loop the 
```suggestion
  const totalPageItems = Math.ceil(props.totalItems / props.itemsPerPage;
  
  return (
  {[...Array(totalPageItems)].map((pageNumber) => console.log(pageNumber))}
  )
```

File: fe/src/pages/BufferPage.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+function BufferPage() {
+  /**
+   * I will implement routing middleware soon using this buffer page
```
Comment (escbooster12-dev) [inline fe/src/pages/BufferPage.jsx]: - what is buffer page?

File: fe/src/components/Pagination.jsx (hunk 1 above)
Comment (ragg09-sunasterisk) [reply fe/src/components/Pagination.jsx]: Hi sir  @escbooster12-dev I tried it again, this is what happened

image
image

File: fe/src/components/Pagination.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+              to={"#"}
+              className="page-link"
+              onClick={() => props.paginateTo(page)}
+            >
+              {page}
+            </Link>
+          </li>
+        ))}
+      </ul>
+    </nav>
```
Comment (escbooster12-dev) [inline fe/src/components/Pagination.jsx]:
```suggestion
    <nav aria-label="Page Navigation">
      <ul className="pagination">
        {[...Array(totalPageItems)].map((page) => (
          <li className="page-item" key={page}>
            <Link
              to={"#"}
              className="page-link"
              onClick={() => props.paginateTo(page)}
            >
              {page}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
```

**Review by escbooster12-dev — APPROVED** [review]

**Pull request merged**